### PR TITLE
Switch PyMoose to use Maturin

### PR DESCRIPTION
**Pull request by jvmncs**
Also
- switches from `setup.py` to `pyproject.toml` w/ maturin as the build-system
- remove some old pymoose release targets in Makefile
- add usage example to pymoose readme and other small updates

### CODE REVIEW

File: pymoose/pymoose/computation/utils.py [hunk 1]
```diff
@@ -84,7 +84,7 @@ def serialize_computation(computation):
 
 
 def deserialize_computation(bytes_stream):
-    computation = msgpack.unpackb(bytes_stream, object_hook=_decode)
+    computation = msgpack.unpackb(bytes_stream, object_hook=_decode, raw=False)
```
Comment (mortendahl) [inline pymoose/pymoose/computation/utils.py:87]: `False` is the default, or does this change anything?

**Review by mortendahl — APPROVED** [review]
nice! 💪

File: pymoose/pymoose/computation/utils.py (hunk 1 above)
Comment (jvmncs) [reply pymoose/pymoose/computation/utils.py:87]: this being the default depends on the version, it wasn't working if I didn't make this explicit

Comment (jvmncs) [reply pymoose/pymoose/computation/utils.py:87]: "wasn't working" as in the objects weren't being decoded properly, everything came out as bytes

**Pull request merged**